### PR TITLE
Ported compiler from Amadeus

### LINKED ISSUE #1: [Amadeus] Migrate Compiler
We need to migrate the Compiler cog from Amadeus to CompSci (Compiler submodule). 

See the cog [here](https://github.com/Czechbol/Amadeus/blob/master/cogs/compiler.py).

The cog doesn't use a database and is using the [Wandbox](https://github.com/melpon/wandbox/blob/master/kennel2/API.rst) API so this will be fairly a easy port.

Feel free to rewrite any part of the cog to make it better/more readable

### PR DESCRIPTION
**Pull request by czechbol**
Resolves #1

### CODE REVIEW

File: compiler/module.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+            None,
+        )
+        for idx, comp in enumerate(comps):
+            if idx % 24 == 0:
+                if idx != 0:
+                    embeds.append(embed)  # noqa F821
+                embed = utils.discord.create_embed(
+                    author=ctx.author,
+                    title=_(ctx, "Language compilers"),
+                    description=f"Templates available: `{lang['templates']}`",
```
Comment (BorekZnovustvoritel) [inline compiler/module.py]: I would replace this line with:
```description=f"Templates available: `{'`, `'.join(lang['templates'])}`",```

**Review by BorekZnovustvoritel — CHANGES_REQUESTED** [review]
One small detail and then I approve.

File: compiler/module.py (hunk 1 above)
Comment (czechbol) [reply compiler/module.py]: This has been resolved

**Review by BorekZnovustvoritel — APPROVED** [review]
All seem good, I approve.

**Pull request merged**